### ISSUE TITLE
[Feature Request] Add Summary Page for Savings Goals

### ISSUE THREAD

**Issue by b4allb4u** [Pool-Of-Tears/GreenStash#161]
### Checklist

- [X] This issue contains only one feature request. I will open one issue for every feature I want to request.
- [X] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo.
- [ ] This feature is merely an UI/UX update.
- [X] This feature is not going to conflict with many of the existing features/options.

### Feature Request: Add Summary Page for Savings Goals
Description:
I would like to request a new feature that provides a dedicated Summary Page for an overview of the user's savings goals within the GreenStash app.

### Describe the solution you'd like

**Feature Details:**
- Page Name: Summary of Savings Goals.
- Display Information:

1.  **Total Number of Active Savings Goals:** Show the number of ongoing savings goals. 
2. **Total Amount Saved:** Display the overall amount saved across all goals.
3. **Current Month's Savings:** Show the total amount saved for the current month.
4. **Upcoming Savings Goals:** List details of goals that are nearing their target date, including goal name, target amount, and days left until the target date.

### How will you/everyone benefit from this feature?

This feature will help users to easily track their savings progress at a glance, see upcoming goals, and stay motivated to achieve their financial goals.

**Comment by starry-shivam** [comment]
I like the idea; however, I will need to come up with a better UI design to display all that information in an organized and visually appealing way. I'll try to implement it in future releases. Thanks!

**Comment by volard** [comment]
@starry-shivam I would like to contribute to your project ^_^ Could you share your vision of "better UI" so I can help with the issue?

**Comment by starry-shivam** [comment]
@volard Thank you for showing interest in contributing to the project. Unfortunately, I have taken an indefinite break from programming and open-source due to various personal reasons and situations happening in my life, that I would rather not discuss. If I ever return to it, I will let you know.

**Comment by volard** [comment]
@starry-shivam sure, no problem! Wish you all the best and hope you'll be fine 👍

**Comment by volard** [comment]
@starry-shivam hey, how it's going? I saw some activity from you recently, so decided to kindly ask if you managed to solve your personal issues and got back to your amazing open source work? 😄

**Comment by starry-shivam** [comment]
> [@starry-shivam](https://github.com/starry-shivam) hey, how it's going? I saw some activity from you recently, so decided to kindly ask if you managed to solve your personal issues and got back to your amazing open source work? 😄

Yes, I did have some free time recently, so I managed to work on a few features and bug fixes. If you want to contribute, sure, you can go ahead. I think I can free up some time now to discuss and review changes.

Regarding the UI idea for the summary, I’m not really sure. We could add a dedicated summary page and call it Home, and then move goals to a separate Goals page. However, that might be a bit too much, since I don’t think the summary would be large enough to consume an entire page.

Instead, we could modify the homepage itself to show a kind of summary card when two or more goals are available. That would be better from both a UI/UX perspective and shouldn’t be too hard to implement. The remaining question is the UI design of such a summary card, I’m not entirely sure about that yet, but I’m open to ideas.
